Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -2525,19 +2561,18 @@ double webkitWebViewBaseGetPageScale(WebKitWebViewBase* webkitWebViewBase)
     return webkitWebViewBase->priv->pageScaleFactor;
 }
 
-static void deviceScaleFactorChanged(WebKitWebViewBase* webkitWebViewBase)
-{
-    webkitWebViewBase->priv->pageProxy->setIntrinsicDeviceScaleFactor(gtk_widget_get_scale_factor(GTK_WIDGET(webkitWebViewBase)));
-    refreshInternalScaling(webkitWebViewBase);
-}
-
 void webkitWebViewBaseCreateWebPage(WebKitWebViewBase* webkitWebViewBase, Ref<API::PageConfiguration>&& configuration)
 {
     WebKitWebViewBasePrivate* priv = webkitWebViewBase->priv;
 
     WebProcessPool& processPool = configuration->processPool();
     priv->pageProxy = processPool.createWebPage(*priv->pageClient, WTFMove(configuration));
+
+#if !GTK_CHECK_VERSION(4, 12, 0)
+    // The fractional scaling value for Gtk 4.12+ is only available
+    // once the widget was realized, so no scaling factor will be set here.
     priv->pageProxy->setIntrinsicDeviceScaleFactor(gtk_widget_get_scale_factor(GTK_WIDGET(webkitWebViewBase)));
```

Comment:
But when not using fractional scaling this is still correct, right? And when realized the surface will return the same value and we'll do nothing. Or am I wrong?

Reply:
What I meant is that we could get the int scale factor always here, so that in case of not fractional scaling, we get is as early as possible. And in case of fractional scaling it's going to change anyway when the view is realized later.